### ISSUE TITLE
Wait for the CMS to be ready before triggering govdelivery

### ISSUE THREAD

**Issue by timcosgrove** [department-of-veterans-affairs/va.gov-cms#7242]
## Description

In the past few days, there have been a number of content release failures related to trying to do stuff while the CMS is deploying.

Here are a couple of examples:

* https://github.com/department-of-veterans-affairs/content-build/actions/runs/1555528870
* https://github.com/department-of-veterans-affairs/content-build/actions/runs/1532069013

Instead of just sending requests, let's wait for the CMS to be ready and _then_ send requests.

## Acceptance criteria
- [x] Content build checks that the CMS is online and ready to receive requests (and waits for that state if needed) before sending graphql requests
	* https://github.com/department-of-veterans-affairs/content-build/pull/851
- [x] Content build checks that the CMS is online and ready to receive requests (and waits for that state if needed) before sending govdelivery callback
	* https://github.com/department-of-veterans-affairs/content-build/pull/832

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

**Comment by timcosgrove** [comment]
Cameron did work here: https://github.com/department-of-veterans-affairs/content-build/pull/832

**Comment by cweagans** [comment]
This is waiting on https://github.com/department-of-veterans-affairs/content-build/pull/851 to be approved/merged. Other than that, our part here is done.

**Comment by cweagans** [comment]
PR merged

**Comment by cweagans** [comment]
For posterity (in case this comes up in critical defect reporting):

The real critical defect in this ticket was AC 1 (wait for the CMS to be ready before triggering govdelivery). Because the build didn't wait for the CMS, it would attempt to trigger govdelivery during the daily deploy of the CMS, which would fail. I opened a PR for that problem within an hour or two of that daily deploy failing. This issue was opened the following day and sized in our morning scrum. The PR that resolves the problem was merged prior to the daily deploy on Dec 9 (> 24 hour turnaround).

During the conversation where we sized this issue and added it to the sprint, we talked about another similar failure case for the content build that happened much less frequently (AC 2). That part wasn't as critical because it didn't happen very often, but because this ticket was sized slightly higher to allow for the possibility of a delay in getting the first PR in, we elected to just toss that part into this ticket as well. A PR for AC 2 was opened on Dec 14. The critical defect label also wasn't added until the 14th. Regrettably, the 14th was the Friday before a bunch of people were out of the office for the holidays, so between OOO and code freeze, that second PR just sat there for a while (we did not have write access to the content build repo at the time). Following the holidays, Tim Wright approved the PR and it was merged the same day.

tldr: The critical part of this defect was resolved immediately. A less critical, preventative change sat for a while over the holidays.
